source/commands/bounty.js

```diff
@@ -0,0 +1,65 @@
+const { PermissionFlagsBits, ActionRowBuilder, StringSelectMenuBuilder } = require('discord.js');
+const { CommandWrapper } = require('../classes');
+const { database } = require('../../database');
+const { getNumberEmoji } = require('../helpers');
+
+const customId = "bounty";
+const options = [];
+const subcommands = [
+	{
+		name: "post",
+		description: "Post your own bounty (+1 XP)",
+		optionsInput: []
+	}
+];
+module.exports = new CommandWrapper(customId, "Bounties are user-created objectives for other server members to complete", PermissionFlagsBits.ViewChannel, false, true, 3000, options, subcommands,
```

Comment:
Something that I'm noticing with this wrapper functionality is that it currently looks/feels overloaded with parameters, which leads to run-on-like lines like this. Reading these parameters is pretty tough, especially without using an IDE that gets tailored to this use case. Probably not something to address in the immediate term, but something to look at improving in the future. In the more immediate future, figuring out clever ways to have better line breaks within this line may help with readability.

Reply:
![image](https://github.com/Imaginary-Horizons-Productions/BountyBot/assets/6503151/c53c4bb3-355b-4822-bd26-c7433e6eae6c)
In that case... cooldown, execute, and isPremium aren't used in the slash command builder, and thus would be good candidates for pulling out of the constructor. The inheritance structure would be pretty empty at that point too

Reply:
Something something KISS rule? Probably an effort for a different pull request.

Reply:
Yeah, Button, Select and Modal all fitting the same "non slash command" payload was a plesaent surprise